sys/page_revisions.js

```diff
@@ -80,6 +83,152 @@ PRS.prototype.getTableSchema = function() {
 };
 
 /**
+ * The name of the suppression table
+ * @type {string}
+ * @const
+ */
+PRS.prototype.restrictionsTableName = 'restrictions';
+
+/**
+ * Suppression table schema
+ *
+ * @type {Object}
+ * @const
+ */
+PRS.prototype.restrictionsTableSchema = function() {
+    return {
+        table: this.tableName,
+        version: 1,
+        attributes: {
+            title: 'string',
+            rev: 'int',
+            restrictions: 'set<string>',
+            page_deleted: 'timeuuid'
+        },
+        index: [
+            { attribute: 'title', type: 'hash' },
+            { attribute: 'rev', type: 'range', order: 'desc' },
+            { attribute: 'page_deleted', type: 'static' }
+        ]
+    };
+};
+
+/**
+ * Returns the suppression table URI for a given domain
+ * @param {string} domain the domain
+ * @returns {URI} suppression table URI
+ */
+PRS.prototype.restrictionsTableURI = function(domain) {
+    return new URI([domain, 'sys', 'table', this.restrictionsTableName, '']);
+};
+
+PRS.prototype.getRestriction = function(hyper, req) {
+    var self = this;
+    var rp = req.params;
+    var attributes = { title: rp.title };
+    if (rp.revision) {
+        attributes.rev = rp.revision;
```

Comment:
If we select `ge` and order by rev asc, then we should be able to get the restriction for the exact revision requested, *or any newer*. If we always store a row on page deletion corresponding to the last revision of that page, this would mean that we don't need to perform a second request on 404, as

- either the page is deleted, in which case we'll get a result (not necessarily for the exact revision we asked for), or
- nothing is found, which means there are no restrictions on either the revision or page.